src/meili/map_matcher.cc

```diff
@@ -220,10 +244,14 @@ MatchResult FindMatchResult(const MapMatcher& mapmatcher,
              next_stateid = time + 1 < stateids.size() ? stateids[time + 1] : StateId();
   const auto& measurement = mapmatcher.state_container().measurement(time);
 
+  // this one couldnt be matched
   if (!stateid.IsValid()) {
     return CreateMatchResult(measurement);
   }
 
+  bool begins_discontinuity = false;
+  bool ends_discontinuity = false;
```

Comment:
since we know in this function which match results begin and end discontinuities we track that here rather than have to figure it out again upstream in thor